x-pack/plugins/ml/public/application/data_frame_analytics/common/get_index_data.ts

```diff
@@ -47,9 +47,12 @@ export const getIndexData = async (
         }, {} as EsSorting);
 
       const { pageIndex, pageSize } = pagination;
+      // TODO: remove results_field from `fields` when possible
```

Comment:
Is this TODO still necessary?

Reply:
The todo is necessary for now because we are getting the flattened `fields.ml.blah` as well as the unflattened `_source.ml` object. Since the `fields` API might change in the future to either return the values unflattened or add a way to exclude certain fields, maybe we can do a follow up PR for when the API changes.